### PR TITLE
New weather type snow

### PR DESCRIPTION
**Pull request by brettpenzer123**
I think this is correct.

### CODE REVIEW

File: src/addresses.h [hunk 1]
```diff
@@ -374,6 +374,8 @@
 #define RCT2_ADDRESS_NEXT_WEATHER_GLOOM				0x013CA751
 #define RCT2_ADDRESS_CURRENT_RAIN_LEVEL				0x013CA752
 #define RCT2_ADDRESS_NEXT_RAIN_LEVEL				0x013CA753
+#define RCT2_ADDRESS_CURRENT_SNOW_LEVEL				0x013CA755
+#define RCT2_ADDRESS_NEXT_SNOW_LEVEL				0x013CA756
```
Comment (IntelOrca) [inline src/addresses.h]: You can't do this, you are writing over new items. Just define these like normal C variables.

This doesn't mean you can not save them to saved games. The only way of overcoming that is using rain level.

File: src/drawing/snow.c [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+* For more information, visit https://github.com/OpenRCT2/OpenRCT2
+*
+* OpenRCT2 is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* A full copy of the GNU General Public License can be found in licence.txt
+*****************************************************************************/
+#pragma endregion**************/
```
Comment (IntelOrca) [inline src/drawing/snow.c]: Remove `**************/`

File: src/addresses.h (hunk 1 above)
Comment (brettpenzer123) [inline src/addresses.h]: Ok.

File: src/world/climate.h [hunk 3]
```diff
@@ -26,20 +26,22 @@ enum {
...
-enum{
+enum {
 	WEATHER_SUNNY,
 	WEATHER_PARTIALLY_CLOUDY,
 	WEATHER_CLOUDY,
 	WEATHER_RAIN,
 	WEATHER_HEAVY_RAIN,
-	WEATHER_THUNDER
+	WEATHER_THUNDER,
+	WEATHER_SNOW
```
Comment (IntelOrca) [inline src/world/climate.h]: Before you can do this, you have to check what happens when you load a saved game that is snowing in vanilla RCT2. It may crash the original game.

File: src/drawing/snow.c [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+	rct_window* newWindow = gWindowNextSlot;
+
+	for (rct_window* w = g_window_list; w < newWindow; w++) {
+		draw_snow_window(w, left, right, top, bottom, draw_snow_func);
+	}
+}
+
+/**
+*
+*  rct2: 0x00684218
```
Comment (duncanspumpkin) [inline src/drawing/snow.c]: None of these functions will have rct2 addresses as they are made by you and not rct2.

File: src/world/climate.h (hunk 3 above)
Comment (brettpenzer123) [inline src/world/climate.h]: It never did snow in vanilla RCT2.

File: src/drawing/snow.c (hunk 4 above)
Comment (brettpenzer123) [inline src/drawing/snow.c]: Nearly all lines of coding in the `snow.c` is C&P from `rain.c`. I've haven't properly cleaned up the file it seems.

File: src/drawing/snow.c (hunk 2 above)
Comment (brettpenzer123) [inline src/drawing/snow.c]: I have no existence of `**************/` within my snow.c file.

File: src/addresses.h (hunk 1 above)
Comment (brettpenzer123) [inline src/addresses.h]: Just out of curiosity why can't I add this information here?

File: src/world/climate.h (hunk 3 above)
Comment (IntelOrca) [inline src/world/climate.h]: 1. Whilst it is snowing, save the game.
2. Load that saved game in vanilla RCT2.
3. Does it crash?

Yes Vanilla RCT2 didn't implement snow which is why you need to check what happens. We still support it and we can't let new features caused saved games to not work in vanilla.

File: src/drawing/snow.c (hunk 2 above)
Comment (IntelOrca) [inline src/drawing/snow.c]: You can see it directly above this comment after the "endregion"

File: src/addresses.h (hunk 1 above)
Comment (IntelOrca) [inline src/addresses.h]: Because... if you look at the address of RCT2_ADDRESS_NEWS_ITEM_LIST, it overlaps your new variables. You cannot insert brand new variables in the region of the original RCT2 data sector!

File: src/drawing/snow.c (hunk 2 above)
Comment (brettpenzer123) [inline src/drawing/snow.c]: I see it above but its not within my file maybe I removed it after pull request.

**Comment by IntelOrca** [comment]
Your changes also fail to build due to a linker error.
See: https://ci.appveyor.com/project/IntelOrca/openrct2-ject9/build/0.0.5.2637

File: src/world/climate.h (hunk 3 above)
Comment (brettpenzer123) [inline src/world/climate.h]: In vanilla RCT2 it just going back to its default weather = `WEATHER_PARTIALLY_CLOUDY`

File: data/language/en-GB.txt [hunk 5]
```diff
@@ -4168,6 +4168,9 @@ STR_5857    :{SMALLFONT}{BLACK}Game options
 STR_5858    :{SMALLFONT}{BLACK}Use GPU for displaying instead of CPU. Improves compatibility with screen capture software. May slightly decrease performance.
 STR_5859    :{SMALLFONT}{BLACK}Enables frame tweening for visually{NEWLINE}smoother gameplay. When disabled,{NEWLINE}the game will run at 40 FPS.
 
+#Additional Strings
```
Comment (Nubbie) [inline data/language/en-GB.txt]: You can remove 'Additinonal string' and leave a space there instead, have no idea why there is a 'tooltip' above but I guess we will let that be :v

File: src/world/climate.h (hunk 3 above)
Comment (IntelOrca) [inline src/world/climate.h]: ok

File: src/world/climate.c [hunk 6]
```diff
@@ -330,13 +340,14 @@ static int climate_play_thunder(int instanceIndex, int soundId, int volume, int
 
 // rct2: 0x00993C94
 // There is actually a sprite at 0x5A9C for snow but only these weather types seem to be fully implemented
-const rct_weather climate_weather_data[6] = {
-	{ .temp_delta = 10, .effect_level = 0, .gloom_level = 0, .rain_level = 0, .sprite_id = 0x5A96 }, // Sunny
```
Comment (janisozaur) [inline src/world/climate.c]: Why did you remove those spaces?

Comment (brettpenzer123) [inline src/world/climate.c]: Reason why I can't?

**Comment by janisozaur** [comment]
please remove 3774290f241c1218908cb86043e5f4023966c

File: src/world/climate.c (hunk 6 above)
Comment (janisozaur) [inline src/world/climate.c]: there's no technical limitation, but it seems an unnecessary change and one that breaks ease of digesting this initialisation.

**Comment by brettpenzer123** [comment]
It's becoming way to complicated so I've scrapped the whole concept.

**Comment by brettpenzer123** [comment]
So I've made the final adjustments. Won't I need to pull request my vcxproj files as they include snow.c into the source. I've also redone ` .temp_delta = 10, .effect_level = 0, .gloom_level = 0, .rain_level = 0, .snow_level = 0, .sprite_id = 0x5A96 `. old ways...

**I feel like I'm stepping on everyone's feet here (cheers for the welcome).**
